test/FunctionalTests/Microsoft.OData.Edm.Tests/Csdl/CsdlWriterTests.cs

```diff
@@ -311,7 +311,8 @@ public void WriteNavigationPropertyInComplexType()
       },
       ""Addresses"": {
         ""$Collection"": true,
-        ""$Type"": ""DefaultNs.Address""
+        ""$Type"": ""DefaultNs.Address"",

```

Comment:
Nullable facet is required for collections in XML, but I don't think it's required in JSON.

Which means our factoring is a bit off -- we shouldn't have code in the serializer to determine whether the null needs to be written. Instead, we should pass down the information that lets the XML serializer decide whether or not to write the null value.